ApollosProject/apollos-apps
# Adds Pull to refetch on Features feed

## DESCRIPTION

### What does this PR do, or why is it needed?

This PR has made a few notable changes to add pull to refetch. 

1. Enables the `FeedView` to accept loading state data directly, rather than having to ignore the `loading` prop. 

2. Adds a `refetchRef` method the FeaturesFeed that collects all the `refetch` functions from each of the children, making sure not to collect any single ref more than once. 

![2020-04-27 17 29 19](https://user-images.githubusercontent.com/1637694/80422740-b4f61500-88ac-11ea-9388-68c8143de05c.gif)


### What design trade-offs/decisions were made?

I'm not in love with the `refetchRef` concept. However, I don't see any better options besides building a `Provider` to handle this, which seems overkill. At least `refetchRef` is based on an established react pattern. Open to suggestions. 

### How do I test this PR?

1. Pull down this PR, link repos, boot the app.
2. Pull down from the home feed, the feed should refetch and you can can confirm in the API console that new data is indeed loading. 

### What automated tests did you write?

New tests where appropriate. 

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
